client/components/file-picker/index.jsx

```diff
@@ -4,27 +4,66 @@
  * External dependencies
  */
 import React from 'react';
-import assign from 'lodash/assign';
 import noop from 'lodash/noop';
-import pick from 'component-file-picker';
 
 export default class FilePicker extends React.Component {
-	constructor( props ) {
-		super( props );
-		this.showPicker = this.showPicker.bind( this );
+
+	getFileAttributes() {
+		return {
+			multiple: this.props.multiple,
+			webkitDirectory: this.props.directory,
+			mozDirectory: this.props.directory,
+			msDirectory: this.props.directory,
+			oDirectory: this.props.directory,
+			directory: this.props.directory,
+			accept: this.props.accept
+		};
+	}
+
+	// This method is only required until React adds directory to their whitelist.
+	// @see https://github.com/facebook/react/issues/3468
+	// Once it is added, `setFileDirectoryAttribute`, `componentDidMount` and
+	// `componentDidUpdate` methods can be removed.
+	// Note: they are already included in the `getFileAttributes` method.
+	setFileDirectoryAttribute() {
+		if ( this.props.directory ) {
+			const fileInput = this.refs.fileInput;
+			fileInput.setAttribute( `webkitdirectory`, '' );
```

Comment:
Good catch. When I initially implemented, I used a _cute_ one-liner to iterate prefixes, like so:

```
'webkit moz ms o '.split(' ').forEach(prefix => fileInput.setAttribute( `${prefix}directory`, '' ));
```

I later decided there was no need for cute overhead (not to be confused with [cuteoverload](http://cuteoverload.com/)) and flattened out the code. 

The remnant ticks are an oversight. Will replace.